### PR TITLE
Implentation for User Login and Signup (Backend)

### PR DESCRIPTION
**Pull request by Ememobong28**
1.) Added routes for user authentication and account creation 
2.) Implemented endpoints for user signup and login 
3.) Verified functionality with successful testing of endpoints using Postman

### CODE REVIEW

**Comment by nicolec2626** [comment]
The description in this PR is good, but in the future you can include a screenshot showing how you tested using Postman. 
Typically in the test plan, we want to see "this is how I tested my code, and this is how I know it works" instead of "I tested my code"

**Comment by nicolec2626** [comment]
It'll also be helpful to get in the habit of separating content into description (what this PR is doing), milestones (which feature this contributes to), and test plan

File: techhive/server.js [hunk 1]
```diff
@@ -39,6 +44,60 @@ app.get('/internships/:id', async (req, res) => {
...
+      res.json(user);
+    } else {
+      res.status(404).json({ message: 'User not found' });
+    }
+  } catch (err) {
+    res.status(500).json({ message: err.message });
+  }
+});
+
+// Route to get all posts, with associated users
```
Comment (nicolec2626) [inline techhive/server.js:71]: Assuming posts here are internship postings, what exactly are associated users? 
How do we define the association? Are they, for example, users that have saved/applied/recommended for a given internship? 

This context would be helpful to include in your PR description

**Comment by Ememobong28** [comment]
Okay noted thank you so much for the feedback

**Comment by Ememobong28** [comment]
> 

Okay taken note of this

File: techhive/server.js (hunk 1 above)
Comment (Ememobong28) [reply techhive/server.js:71]: I need to make a few edits to this, users here are the students and company login into TechHive or creating accounts. Okay sure Ill work on more PR descriptions

**Pull request merged**